include/slam_toolbox/system_utils.hpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * system_utils
+ * Copyright (c) 2019, Samsung Research America
+ *
+ * THE WORK (AS DEFINED BELOW) IS PROVIDED UNDER THE TERMS OF THIS CREATIVE
+ * COMMONS PUBLIC LICENSE ("CCPL" OR "LICENSE"). THE WORK IS PROTECTED BY
+ * COPYRIGHT AND/OR OTHER APPLICABLE LAW. ANY USE OF THE WORK OTHER THAN AS
+ * AUTHORIZED UNDER THIS LICENSE OR COPYRIGHT LAW IS PROHIBITED.
+ *
+ * BY EXERCISING ANY RIGHTS TO THE WORK PROVIDED HERE, YOU ACCEPT AND AGREE TO
+ * BE BOUND BY THE TERMS OF THIS LICENSE. THE LICENSOR GRANTS YOU THE RIGHTS
+ * CONTAINED HERE IN CONSIDERATION OF YOUR ACCEPTANCE OF SUCH TERMS AND
+ * CONDITIONS.
+ *
+ */
+
+/* Author: Steven Macenski */
+
+#ifndef SLAM_TOOLBOX__SYSTEM_UTILS_HPP_
+#define SLAM_TOOLBOX__SYSTEM_UTILS_HPP_
+
+#ifndef _WIN32
+  #include <sys/resource.h>
+#endif
+
+namespace system_utils
+{
+
+inline void setStackLimitMaximum(size_t stack_size)
+{
+#ifndef _WIN32
```

Comment:
So will this not work if on windows?

Reply:
The Windows and Linux are using different ways to ask the underlaying runtime libraries to reserve the stack space for the threads in a process. On Windows, the programmatical way is to pass the reserved (maximum) stack size when [CreateThread](https://docs.microsoft.com/en-us/windows/win32/procthread/thread-stack-size) is actually getting called. Currently I haven't deep dived into where those points occurring in the `slam_toolbox`, so yes, this pull request doesn't yet bring up this on Windows. And any pointers to what's thread stack to be guarded by this variable would be helpful for me to look into that later. :)

Reply:
So in windows-land you can't increase the stack memory for the process? That's a pretty important thing for this project since it is critically necessary for deserializing and serializing maps. This is done in the main thread of the process, not a spin off.

If that's the case, we should add a note in the readme under the relevant sections saying it isn't supported on windows.